### PR TITLE
feat(TextInputIcon): Added optional align prop which can be set to left and defaults to right

### PR DESCRIPTION
**Pull request by benji6**
screen shot 2017-08-03 at 10 27 43
https://feat-textinputicon-align-left--axiom-rules.netlify.com/docs/components/form

### CODE REVIEW

File: src/components/form/TextInputIcon.js [hunk 1]
```diff
@@ -1,19 +1,29 @@
+import classnames from 'classnames';
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import { Icon, Link } from 'bw-axiom';
 
 export default class TextInputIcon extends Component {
   static propTypes = {
+    align: PropTypes.oneOf(['left', 'right']).isRequired,
```
Comment (thatguynamedandy) [inline src/components/form/TextInputIcon.js]: You could probably drop `isRequired` as it has a default.

Comment (benji6) [inline src/components/form/TextInputIcon.js]: yeah I'm not sure if that even does anything if there are defaults! I'll remove it :D

File: src/components/form/TextInput.css [hunk 2]
```diff
@@ -33,6 +33,10 @@
   }
 }
 
+.ax-input__icon--align-left {
+  order: -1;
```
Comment (HHogg) [inline src/components/form/TextInput.css]: ❤️

File: src/components/form/TextInputIcon.js [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
+    const className = classnames('ax-input__icon', {
+      [`ax-input__icon--align-${align}`]: align,
+    });
 
     return (
-      <span className="ax-input__icon">
-        <Link { ...this.props } style="subtle">
-          <Icon name={ name }  size="1rem" />
+      <span className={ className }>
+        <Link { ...rest } style="subtle">
```
Comment (HHogg) [inline src/components/form/TextInputIcon.js]: I guess the Link part should now be conditional? Should the TextInputIcons on the left appear clickable?

Comment (HHogg) [inline src/components/form/TextInputIcon.js]: Yeah the `onClick` would also be consistent with other components like Bars and DataPoints.

File: src/components/form/TextInput.css (hunk 2 above)
Comment (benji6) [inline src/components/form/TextInput.css]: <3 flexbox!

File: src/components/form/TextInputIcon.js (hunk 1 above)
Comment (HHogg) [inline src/components/form/TextInputIcon.js]: Yeah I'm not sure about warning but removing the `isRequired` will be correct for the API docs :)

**Pull request merged**